### PR TITLE
[CIS-404] Expose message replies on MessageController

### PR DESCRIPTION
**Pull request by mantergo**
**In this PR:**

- Expose `latestReplies` on `ChatMessage`
- Expose `replies` on `MessageController`

### CODE REVIEW

File: Sources_v3/Controllers/MessageController/MessageController.swift [hunk 1]
```diff
@@ -179,6 +222,59 @@ public extension _ChatMessageController {
...
+        after messageId: MessageId? = nil,
+        limit: Int = 25,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        let lastMessageId = messageId ?? replies.last?.id
+    
+        messageUpdater.loadReplies(
+            cid: cid,
+            messageId: self.messageId,
+            pagination: MessagesPagination(pageSize: limit, parameter: lastMessageId.map { PaginationParameter.lessThan($0) })
```
Comment (VojtaStavik) [inline Sources_v3/Controllers/MessageController/MessageController.swift:245]: Are "next" replies `lessThen` id? 🤔

File: Sources_v3/Controllers/MessageController/MessageController.swift [hunk 2]
```diff
@@ -179,6 +222,59 @@ public extension _ChatMessageController {
...
+        guard let messageId = messageId ?? replies.first?.id else {
+            log.error(ClientError.MessageEmptyReplies().localizedDescription)
+            callback { completion?(ClientError.MessageEmptyReplies()) }
+            return
+        }
+    
+        messageUpdater.loadReplies(
+            cid: cid,
+            messageId: self.messageId,
+            pagination: MessagesPagination(pageSize: limit, parameter: .greaterThan(messageId))
```
Comment (VojtaStavik) [inline Sources_v3/Controllers/MessageController/MessageController.swift:273]: I think there should be `lesttThen` here.

File: Sources_v3/Controllers/MessageController/MessageController+Combine.swift [hunk 3]
```diff
@@ -17,6 +17,11 @@ extension _ChatMessageController {
         basePublishers.messageChange.keepAlive(self)
     }
     
+    /// A publisher emitting a new value every time the list of the replies the message has changes.
```
Comment (VojtaStavik) [inline Sources_v3/Controllers/MessageController/MessageController+Combine.swift:20]: ... the list of the replies **of** the message has changes...

File: Sources_v3/Controllers/ChannelController/ChannelController_Tests.swift [hunk 4]
```diff
@@ -267,9 +267,6 @@ class ChannelController_Tests: StressTestCase {
         // Set bottom-to-top ordering
         controller.listOrdering = .bottomToTop
         
-        // Simulate `synchronize` call to apply changes
-        controller.synchronize()
-
```
Comment (VojtaStavik) [inline Sources_v3/Controllers/ChannelController/ChannelController_Tests.swift:272]: Why this change?

**Review by VojtaStavik — APPROVED** [review]
Great work, thanks!

File: Sources_v3/Controllers/ChannelController/ChannelController_Tests.swift (hunk 4 above)
Comment (mantergo) [reply Sources_v3/Controllers/ChannelController/ChannelController_Tests.swift:272]: This test was written when there were no actions on `listOrdering` `didSet`.
But later we added DB observer reset on setting `listOrdering` so no need for `synchronize()` call.

File: Sources_v3/Controllers/MessageController/MessageController.swift (hunk 1 above)
Comment (mantergo) [reply Sources_v3/Controllers/MessageController/MessageController.swift:245]: Fixed in ae4e119 along with `ChannelController`pagination in 7f9ac07.

File: Sources_v3/Controllers/MessageController/MessageController.swift (hunk 2 above)
Comment (mantergo) [reply Sources_v3/Controllers/MessageController/MessageController.swift:273]: Fixed in ae4e119 along with `ChannelController`pagination in 7f9ac07.

File: Sources_v3/Controllers/ChannelController/ChannelController_Tests.swift (hunk 4 above)
Comment (VojtaStavik) [reply Sources_v3/Controllers/ChannelController/ChannelController_Tests.swift:272]: I see. Please feel free to put a comment like this to the commit body, it's hard to reason about these things without context.

Comment (mantergo) [reply Sources_v3/Controllers/ChannelController/ChannelController_Tests.swift:272]: Sure, will do that next time 👌
